runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -7183,6 +7183,56 @@ done:;
 		return rc;
 	}
 
+	/*
+	 * Determine if the two objects are substitutable
+	 *
+	 * @param[in] lhs the lhs object of acmp bytecodes
+	 * @param[in] rhs the rhs object of acmp bytecodes
+	 * return true if they are substituable and false otherwise
+	 */
+	VMINLINE bool
+	acmp(j9object_t lhs, j9object_t rhs)
+	{
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+		bool acmpResult = false;
+		if (rhs == lhs) {
+			acmpResult = true;
+		} else if ((NULL == rhs) || (NULL == lhs)) {
+			acmpResult = false;
```

Comment:
This could be optimized given the initial value is false, but I'm not sure how much value there is there (I really hope the compiler would just do the right thing here).